diff --git a/src/org/recompile/freej2me/FreeJ2ME.java b/src/org/recompile/freej2me/FreeJ2ME.java
index c9f7b13..7819664 100644
--- a/src/org/recompile/freej2me/FreeJ2ME.java
+++ b/src/org/recompile/freej2me/FreeJ2ME.java
@@ -86,6 +86,12 @@ public class FreeJ2ME
 		{
 			jarfile = args[0];
 		}
+		if(args.length==2)
+		{
+			Mobile.setPlatform(new MobilePlatform(lcdWidth, lcdHeight));
+			Mobile.getPlatform().jarMetadata(jarfile, args[1]);
+			System.exit(0);
+		}
 		if(args.length>=3)
 		{
 			lcdWidth = Integer.parseInt(args[1]);
diff --git a/src/org/recompile/mobile/MIDletLoader.java b/src/org/recompile/mobile/MIDletLoader.java
index 41a176f..0c27d1e 100644
--- a/src/org/recompile/mobile/MIDletLoader.java
+++ b/src/org/recompile/mobile/MIDletLoader.java
@@ -206,7 +206,6 @@ public class MIDletLoader extends URLClassLoader
 				line = lines.get(i);
 				if(line.startsWith("MIDlet-1:"))
 				{
-					System.out.println(line);
 					line = line.substring(9);
 					parts = line.split(",");
 					if(parts.length == 3)
@@ -237,6 +236,86 @@ public class MIDletLoader extends URLClassLoader
 		}
 	}
 
+	public void getNameFromManifest()
+	{
+		String resource = "META-INF/MANIFEST.MF";
+		URL url = findResource(resource);
+		if(url==null)
+		{
+			resource = "meta-inf/MANIFEST.MF";
+			url = findResource(resource);
+			if(url==null)
+			{
+				resource = "META-INF/manifest.fm";
+				url = findResource(resource);
+				if(url==null)
+				{
+					resource = "meta-inf/manifest.fm";
+					url = findResource(resource);
+					if(url==null)
+					{
+						return;
+					}	
+				}	
+			}
+		}
+
+		String line;
+		String[] parts;
+		int split;
+		String key;
+		String value;
+		try
+		{
+			InputStream is = url.openStream();
+			BufferedReader br = new BufferedReader(new InputStreamReader(is));
+			
+			ArrayList<String> lines = new ArrayList<String>();
+			while ((line = br.readLine()) != null) 
+			{
+				if(line.startsWith(" "))
+				{
+					line = lines.get(lines.size()-1) + line.trim();
+					lines.remove(lines.size()-1);
+				}
+				lines.add(line);
+			}
+
+			for (int i=0; i<lines.size(); i++)
+			{
+				line = lines.get(i);
+				if(line.startsWith("MIDlet-1:"))
+				{
+					line = line.substring(9);
+					parts = line.split(",");
+					if(parts.length == 3)
+					{
+						name = parts[0].trim();
+						System.out.println(name);
+						icon = parts[1].trim();
+						className = parts[2].trim();
+						suitename = name;
+					}
+					//System.out.println("Loading " + name);
+				}
+
+				split = line.indexOf(":");
+				if(split>0)
+				{
+					key = line.substring(0, split).trim();
+					value = line.substring(split+1).trim();
+					properties.put(key, value);
+				}
+			}
+			// for RecordStore, remove illegal chars from name
+			suitename = suitename.replace(":","");
+		}
+		catch (Exception e)
+		{
+			System.out.println("Can't Read Jar Manifest!");
+			e.printStackTrace();
+		}
+	}
 
 	public InputStream getResourceAsStream(String resource)
 	{
diff --git a/src/org/recompile/mobile/MobilePlatform.java b/src/org/recompile/mobile/MobilePlatform.java
index 4fc4501..f744df2 100644
--- a/src/org/recompile/mobile/MobilePlatform.java
+++ b/src/org/recompile/mobile/MobilePlatform.java
@@ -167,6 +167,24 @@ public class MobilePlatform
 
 	}
 
+	public void jarMetadata(String jarurl, String metadatatype)
+	{
+		if(metadatatype.equals("name"))
+		{
+			try
+			{
+				URL jar = new URL(jarurl);
+				loader = new MIDletLoader(new URL[]{jar});
+				loader.getNameFromManifest();
+			}
+			catch (Exception e)
+			{
+				System.out.println(e.getMessage());
+				e.printStackTrace();
+			}
+		}
+	}
+
 	public void runJar()
 	{
 		try
